src/Common/Core/Impl/Extensions/TaskExtensions.cs

```diff
@@ -122,8 +122,16 @@ public static class TaskExtensions {
         /// within the timeout specified.
         /// </summary>
         public static T WaitTimeout<T>(this Task<T> task, int msTimeout) {
-            task.Wait(msTimeout);
-            return task.IsCompleted ? task.Result : default(T);
+            bool result = false;
+            try {
+                task.Wait(msTimeout);
+                result = task.IsCompleted;
+            } catch(AggregateException) { }
```

Comment:
Ah, I see. My earlier remarks about this method were rather dealing with the case where the task itself throws - then you only find out in `Result` (and it'll be an `AggregateException`).

The old code was calling `WaitAndUnwrapExceptions`, which unwraps AE. The new one needs to do so as well to avoid silent changes and possibly breakage (if nothing ends up catching it).